cantools/database/can/signal.py

```diff
@@ -305,6 +321,41 @@ def __init__(self,
             # multi-lingual dictionary
             self.comments = comment
 
+    @property
+    def initial(self) -> Optional[Union[int, SignalValueType]]:
+        warnings.warn(
+            "The `Signal.initial` attribute is deprecated and scheduled for removal in April 2024. "
+            "Use `Signal.raw_initial` or `Signal.scaled_initial` instead.",
+            DeprecationWarning,
+        )
+        return self._initial
+
+    @initial.setter
+    def initial(self, value: Union[int, SignalValueType]) -> None:
+        warnings.warn(
+            "The `Signal.initial` attribute is deprecated and scheduled for removal in April 2024. "
+            "Use `Signal.raw_initial` or `Signal.scaled_initial` instead.",
+            DeprecationWarning,
+        )
+        self._initial = value
+
+    def raw_to_scaled(self, raw: int) -> SignalValueType:
+        with contextlib.suppress(KeyError, TypeError):
+            return self.choices[raw]  # type: ignore[index]
+
+        return raw * self.scale + self.offset
+
+    def scaled_to_raw(self, scaled: SignalValueType) -> int:
+        if isinstance(scaled, (float, int)):
+            _transform = float if self.is_float else round
+            if self.offset == 0 and self.scale == 1:
+                # treat special case to avoid introduction of unnecessary rounding error
+                return _transform(scaled)  # type: ignore[operator,no-any-return]
+
+            return _transform((scaled - self.offset) / self.scale)  # type: ignore[operator,no-any-return]
+
+        return self.choice_string_to_number(str(scaled))
```

Comment:
```suggestion
        if isinstance(scaled, (str, NamedSignalValue)):
            return self.choice_string_to_number(str(scaled))
        raise NotImplemented(f'Conversion of physical values of type "{type(scaled)}" to raw data')
```

Reply:
Done, but with TypeError

Reply:
weird. how come?

Reply:
Why weird? 😄 

The python docs say: 

> Passing arguments of the wrong type (e.g. passing a [list](https://docs.python.org/3/library/stdtypes.html#list) when an [int](https://docs.python.org/3/library/functions.html#int) is expected) should result in a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), but passing arguments with the wrong value (e.g. a number outside expected boundaries) should result in a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError).

Reply:
I misinterpreted you: I thought that mypy complained about the new construct, while you probably said that you raise a `TypeError` instead of `NotImplemented`.